internal/resources/google/storage_bucket.go

```diff
@@ -0,0 +1,233 @@
+package google
+
+import (
+	"github.com/infracost/infracost/internal/resources"
+	"github.com/infracost/infracost/internal/schema"
+	"github.com/infracost/infracost/internal/usage"
+
+	"fmt"
+	"strings"
+
+	"github.com/shopspring/decimal"
+)
+
+type StorageBucket struct {
+	Address                     string
+	Region                      string
+	Location                    string
+	StorageClass                string
+	StorageGB                   *float64                         `infracost_usage:"storage_gb"`
+	MonthlyClassAOperations     *int64                           `infracost_usage:"monthly_class_a_operations"`
+	MonthlyClassBOperations     *int64                           `infracost_usage:"monthly_class_b_operations"`
+	MonthlyDataRetrievalGB      *float64                         `infracost_usage:"monthly_data_retrieval_gb"`
+	MonthlyEgressDataTransferGB *StorageBucketNetworkEgressUsage `infracost_usage:"monthly_egress_data_transfer_gb"`
+}
+
+var StorageBucketUsageSchema = []*schema.UsageItem{
+	{Key: "storage_gb", ValueType: schema.Float64, DefaultValue: 0},
+	{Key: "monthly_class_a_operations", ValueType: schema.Int64, DefaultValue: 0},
+	{Key: "monthly_class_b_operations", ValueType: schema.Int64, DefaultValue: 0},
+	{Key: "monthly_data_retrieval_gb", ValueType: schema.Float64, DefaultValue: 0},
+	{
+		Key:          "monthly_egress_data_transfer_gb",
+		ValueType:    schema.SubResourceUsage,
+		DefaultValue: &usage.ResourceUsage{Name: "monthly_egress_data_transfer_gb", Items: StorageBucketNetworkEgressUsageSchema},
+	},
+}
+
+func (r *StorageBucket) PopulateUsage(u *schema.UsageData) {
+	resources.PopulateArgsWithUsage(r, u)
+	if r.MonthlyEgressDataTransferGB == nil {
+		r.MonthlyEgressDataTransferGB = &StorageBucketNetworkEgressUsage{}
+	}
+}
+
+func (r *StorageBucket) BuildResource() *schema.Resource {
+	region := r.Region
+	components := []*schema.CostComponent{
+		dataStorage(r.Location, r.StorageClass, r.StorageGB),
+	}
+	data := dataRetrieval(r)
+	if data != nil {
+		components = append(components, data)
+	}
+	components = append(components, operations(r.StorageClass, r.MonthlyClassAOperations, r.MonthlyClassBOperations)...)
+
+	r.MonthlyEgressDataTransferGB.Region = region
+	r.MonthlyEgressDataTransferGB.Address = "Network egress"
+	r.MonthlyEgressDataTransferGB.PrefixName = "Data transfer"
+	return &schema.Resource{
+		Name:           r.Address,
+		CostComponents: components,
+		SubResources: []*schema.Resource{
+			r.MonthlyEgressDataTransferGB.BuildResource(),
+		}, UsageSchema: StorageBucketUsageSchema,
+	}
+}
+
+func getDSRegionResourceGroup(location, storageClass string) (string, string) {
```

Comment:
Can this be a method on the struct?

Reply:
I'm not sure cause it's used by `dataStorage` function which is used by both `storage_bucket` and `container registry`. What you think?

Reply:
👍